Jim dev

jsharples777 opened the pull request:
Setting up project for a basic level of JSX processing (not React level) but enough for easier use of html in the code.

Converted ShoppingList to jsx using HTML code and it is processed into js using the babel transpiler.

Updated TestController to use the handle methods to test the functionality better.

Removed the now redundant methods in DOMUtils, using Jsx to do this easier and more efficiently

Added two new script commands and development dependencies in package.json to use babel to convert the JSX to JS.

jsharples777 committed:
Working on using JSX for components

jsharples777 committed:
Working on using JSX for components

jsharples777 committed:
Setting up project for a basic level of JSX processing (not React level) but enough for easier use of html in the code.

Converted ShoppingList to jsx using HTML code and it is processed into js using the babel transpiler.

Updated TestController to use the handle methods to test the functionality better.

Removed the now redundant methods in DOMUtils, using Jsx to do this easier and more efficiently

Added two new script commands and development dependencies in package.json to use babel to convert the JSX to JS.

jsharples777 reviewed (commented):
Control file for the Babel preprocessor for converting basic JSX to JS files

raven-bootcamp reviewed (approved):
Happy to approve this for now, but a lot of it is a bit beyond me, especially revolving around babel, etc.

jsharples777 committed:
Working on using JSX for components

jsharples777 committed:
1. Ensure that the recipe can't be favourited more than once
2. Added a local buffer for future optimization of local data storage (optional)

jsharples777 committed:
1.  Simplified modal show/hide event handling
2.  Started favourite recipes
3.  Start recipe details
